### PR TITLE
feat: kosko deploy with gitlab-deploy image

### PR DESCRIPTION
**Pull request by revolunet**
replace dependency to nok8s docker image with [ci-deploy image](https://github.com/SocialGouv/docker/tree/master/ci-deploy)

 - nok8s version is maintained by renovate
 - kosko dependencies are download at runtime, based on project dependencies

this remove dependencies to docker images nk8s and kosko 

one can use a specific nok8s version in its pipeline with the variable `NOK8S_VERSION`

### CODE REVIEW

File: base_deploy_kosko_stage.yml [hunk 1]
```diff
@@ -34,23 +36,26 @@ variables:
     - |
       if [[ -d "${CI_PROJECT_DIR}/.socialgouv" ]]; then
         echo "Use .socialgouv config."
-        rm -rf /k8s/.socialgouv
-        cp -r ${CI_PROJECT_DIR}/.socialgouv /k8s/
-        if [[ -d "/k8s/.socialgouv/environments" ]]; then
-          cp -r /k8s/.socialgouv/environments/* /k8s/.k8s/environments/
+        echo "Use no-k8s version $NOK8S_VERSION"
```
Comment (douglasduteil) [inline base_deploy_kosko_stage.yml:39]:
```suggestion
        echo "Using no-k8s version $NOK8S_VERSION"
```

**Review by douglasduteil — APPROVED** [review]
LGTM :)

**Review by gary-van-woerkens — APPROVED** [review]

**Comment by revolunet** [comment]
needs update with new autodevops template

**Comment by revolunet** [comment]
obsolete

**Pull request closed without merging**